qemu/tests/rng_hotplug.py

```diff
@@ -122,28 +130,36 @@ def run_subtest(sub_test):
                                        {'id': 'virtio-rng-pci-%d' % num})
             hotplug_rng(vm, new_dev)
             dev_list.append(new_dev)
+            # temporary workaround for migration
+            vm.params["virtio_rngs"] += ' rng%d' % num
+            vm.params["backend_rng%d" % num] = 'rng-random'
 
         # run rng test after hotplug
         restart_rngd(vm)
         run_subtest(rng_basic_test)
 
-        # run reboot/shutdown after hotplug
-        if pm_test_after_plug:
-            run_subtest(pm_test_after_plug)
+        # run reboot/shutdown/migration after hotplug
+        if sub_test_after_plug:
+            run_subtest(sub_test_after_plug)
             # run rng test after reboot,skip followed test if
-            # pm_test_after_plug is shutdown
+            # sub_test_after_plug is shutdown
             if vm.is_alive():
                 run_subtest(rng_basic_test)
             else:
                 return
-
-        stop_rngd(vm)
-        time.sleep(5)
-        for dev in dev_list:
-            unplug_rng(vm, dev)
-
-        # run reboot/shutdown test after hot-unplug
-        if pm_test_after_unplug:
-            run_subtest(pm_test_after_unplug)
+        if sub_test_after_plug == 'migration':
+            # Unplug attached rng device
+            unplug_all_rngs(vm)
+            # temporary workaround for migration
+            vm.params["virtio_rngs"] = ''
+        else:
+            stop_rngd(vm)
+            time.sleep(5)
+            for dev in dev_list:
+                unplug_rng(vm, dev)
+
```

Comment:
Why use two different unplug methods here? What's the diff between unplug_all_rngs(vm) and for dev in dev_list: unplug_rng(vm, dev)?
Is there any way to make it one general function?

Reply:
It's temporary workaround for defects of hotplug/unplug/migration functions. Thanks!